### PR TITLE
chore: simplify initial navigation

### LINKED ISSUE #701: Initial navigation when using region navigation can be simplified by setting IsDefault on Main route
## What would you like to be added:

Instead of calling Navigate from within ShellViewModel/Model, set the IsDefault property on the Main route.

## Why is this needed:

Currently initial navigation is done using an async method from the constructor in ShellViewModel/Model. This isn't great because it's discarding the task and not a pattern that should be encouraged

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): closes #701

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?

Navigate is called from ShellViewModel/Model

## What is the new behavior?

IsDefault is set on the Main route

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ X ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/ShellModel.cs [hunk 1]
```diff
@@ -25,12 +25,8 @@ private async void LoggedOut(object? sender, EventArgs e)
...
     private readonly IAuthenticationService _authentication;
 #else
-        _ = Start();
-    }
-
-    public async Task Start()
-    {
-        await _navigator.NavigateViewModelAsync<MainModel>(this);
+        // Add code here to initialize or attach event handlers to singleton services
+        // for example the LoggedOut event of an authentication service.
```
Comment (dansiegel) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/ShellModel.cs:29]: Not sure that this example will make sense for someone who isn't using authentication.

Comment (nickrandolph) [reply src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/ShellModel.cs:29]: Perhaps, happy for you to think of alternative text ;-)

Comment (nickrandolph) [reply src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/ShellModel.cs:29]:
```suggestion
```

File: src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/ShellViewModel.cs [hunk 2]
```diff
@@ -30,12 +30,8 @@ private async void LoggedOut(object? sender, EventArgs e)
...
     }
 #else
-        _ = Start();
-    }
-
-    public async Task Start()
-    {
-        await _navigator.NavigateViewModelAsync<MainViewModel>(this);
+        // Add code here to initialize or attach event handlers to singleton services
+        // for example the LoggedOut event of an authentication service.
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp.1/Presentation/ShellViewModel.cs:34]:
```suggestion
```

**Review by dansiegel — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Pull request merged**